### PR TITLE
build: Add a gen-github-release.sh script for generating GitHub release contents

### PR DESCRIPTION
**Pull request by gibson042**
Ref #9079

## Description

Add a script for producing release text like that seen at https://github.com/Agoric/agoric-sdk/releases .

### Security Considerations

n/a

### Scaling Considerations

n/a

### Documentation Considerations

Referenced with an example in MAINTAINERS.md.

### Testing Considerations

n/a

### Upgrade Considerations

n/a

### CODE REVIEW

File: scripts/gen-github-release.sh [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+- #$ISSUE: $DESCRIPTION
+
+The release contains at least the following fixes:
+
+- #$ISSUE: $DESCRIPTION
+
+The full set of changes in this release can be found at https://github.com/Agoric/agoric-sdk/pull/$RELEASE_PR.
+
+This code has satisfied all pre-release/testnet validation checks, and is now recommended for nodes to upgrade from the previous `$PREV_RELEASE` release. As a state-machine-breaking upgrade, once approved, all chain validators will need to upgrade from `$PREV_RELEASE` to this new version (after the chain halts due to reaching the height required in a governance proposal).
```
Comment (JimLarson) [inline scripts/gen-github-release.sh]: Change "state-machine-breaking" to "chain-halting", as that's the essential property. It's conceivable that we might want a chain-halting upgrade that's not necessarily state machine breaking.

File: scripts/gen-github-release.sh [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Since the `agoric-upgrade-11` release, state-sync snapshots include more data than before. Nodes which have inadvertently pruned this data (e.g. those created from a state-sync before the `agoric-upgrade-11` release) will not be able to produce such snapshots, and will need to be restored from state-sync. We are aware of continued performance issues related to state-sync. In particular, we've observed that on some deployments, the current implementation can require 100 GB of temporary free disk space and 16GB of memory.
+
+Below is the _cosmos upgrade handler name_ for this release. This is the name that can be used in governance proposals to deploy this upgrade.
+
+```
+Cosmos Upgrade Handler Name: $TAG
+```
+
+Below is the git information related to this software release. Note the _git tag_ and _cosmos upgrade handler name_ differ.
```
Comment (JimLarson) [inline scripts/gen-github-release.sh]: The git tag and cosmos upgrade handler name **may** differ. (They can also be the same.)

File: scripts/gen-github-release.sh [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+yarn build
+(cd packages/cosmic-swingset && make)
+# (start the agd service)
+```
+
+Do _not_ copy the `agd` script or Go binary to another location. If you would like to have an executable `agd` in another location, then create a symlink in that location pointing to `agoric-sdk/bin/agd`.
+
+### Golang Version
+
+The `$TAG` release requires Go 1.20.2 or higher.
```
Comment (JimLarson) [inline scripts/gen-github-release.sh]: This Go minimal version number is getting replicated in too many places. Consider referring to, e.g. README.md at the tag version. Same comment for node and C compiler versions.

**Review by JimLarson — APPROVED** [review]
Just a few optional suggestions on the text.

**Review by iomekam — APPROVED** [review]

**Pull request merged**